### DISCUSSION TITLE
Asterisk PBX : Object not available for viewing - JSON

### DISCUSSION THREAD

**Discussion by omalisson** [grafana/grafana-infinity-datasource#214] [Specific APIs] [answered]
Hi!

I am using Asterisk PBX API to extract some information for my dashboard. I currently use it with mariadb but I have a greater effort to transform it, with Infinity everything is simpler!
Congratulations on the fantastic project!

In my case, I have an impasse in creating cards with the "Stat" view, because when I call the url of a single object (extension) I have no result. When I call the url that returns all the extensions, I have a result on screen, but what I need is one card per branch.
Below are the commented screens:

- Screenshot with the result of url with all extensions:
![image](https://user-images.githubusercontent.com/35525083/146442364-83c5e18e-3601-4ec9-beca-94d955c424c2.png)

- Screenshot with the result of a single extension:
![image](https://user-images.githubusercontent.com/35525083/146442510-c63cd087-b097-4c7a-840f-3e63937eb86f.png)

- Screenshot using grafana's Stat view with all extensions:
![image](https://user-images.githubusercontent.com/35525083/146442854-daf3a466-d3fe-4ee1-a42d-9f0f0a905a5a.png)

- Screenshot using grafana Stat view with selecting single extension:
![image](https://user-images.githubusercontent.com/35525083/146443381-3dcfc435-0627-4a12-96de-21341ada9d53.png)

The scenario of this last print is the same as the second print, but it doesn't work.
I would like to know how to filter a single extension using the url that returns all extensions. I tried something with regex but it didn't work.

I am very grateful to anyone who can help with any information or procedure to resolve this case.

**Comment by yesoreyeram** [comment, marked as answer]
Stats view is for numbers and metrics. In your case, table view should work. Have you tried table panel?

**Reply by omalisson** [reply]
Hello yesoreyeram!

Thanks for your answer!

Regarding the type of visualization, in fact using it as a table works well, but for what I intend to create I need it to work with stat.
Infinity is working as I showed in the third print with all the extensions, but when I want to populate it with just one extension with it also has the json output as shown in the second print it's not working.
If at least I have a regex that I can use in the third print scenario, it would help me, as it would create the number of cards I need, one for each branch.

**Reply by yesoreyeram** [reply]
The role of the infinity is just to fetch the data which it does correctly. So I can this is a feature request for grafana if specific type of data is not working with specific type of panels. ( AFAIK, stats panel works only with timeseries data. so, Changing the format to timeseries may help but not sure ).
